### ISSUE TITLE
Represent links to bad linecodes as clickable links (like flake8 does)

### ISSUE THREAD

**Issue by 57an** [psf/black#2811]
Hi, 
we use black in our CI in pre-commit with --check and --diff options

If there are bad from black point of view lines it report them like something like
...
         result = {'own_key': own_key, 'defended': defended_object, 'substation': substation, 'switches': switches}
         return result
would reformat source\pf\relay_switches.py
Oh no! \U0001f4a5 \U0001f494 \U0001f4a5
1 file would be reformatted.
flake8...................................................................Failed
...
I suggest to make **source\pf\relay_switches.py** linkable to have an ability to walk to this file in one click. Flake8 works this way and this is much more convenient.

Additional enhancement if use links instead of raw text is to visually mark bad files if more then one file reported by black (it is hard to find them now in our usage case)

**Comment by JelleZijlstra** [comment]
How would this work? Does flake8 use some special terminal escape sequences?

**Comment by Shivansh-007** [comment]
Yeah, there are special terminal escape sequences to hyperlink text in the terminal though it is not supported in legacy windows:

```py
In [1]: f"\x1b]8;id=987;https://github.com\x1b\\TEXT\x1b]8;;\x1b\\"
Out[1]: '\x1b]8;id=987;https://github.com\x1b\\TEXT\x1b]8;;\x1b\\'

In [2]: print(_)
TEXT
```

##### See the dot-underlining

![Screenshot from 2022-01-26 11-55-16](https://user-images.githubusercontent.com/69356296/151113950-97b7a1f1-29f7-4aa2-854e-1abdd722e16e.png)



##### Basically:
```py
f"\x1b]8;id={randint(0, 99999)};{link}\x1b\\{text}\x1b]8;;\x1b\\"
```

**Comment by 57an** [comment]
I've studied flake8 source code, and it looks like adding just : and row number gives what I need in this issue. Try in PyCharm IDE code like this:
print(r'D:\py\flake8-main\src\flake8\api\legacy.py:1')

As far as flake8 points concrete line and column of code, this format is suitable.
But black points one file link for all file warnings. 
I think, adding :1 into file links by default is not really good idea (maybe there are external parsers for black output which will be broken with this). 
But adding something like --clickable option into cli may be a good solution.

**Comment by Shivansh-007** [comment]
> As far as flake8 points concrete line and column of code, this format is suitable. But black points one file link for all file warnings.

I agree with this here, but I don't think all terminals are supporting this, nor does the default gnome-terminal, not sure about the others. If we are just wanting to make the file clickable I would go with the former approach of using terminal escape codes, as it is a more guaranteed method that would work on all systems.

![Screenshot from 2022-01-27 06-25-23](https://user-images.githubusercontent.com/69356296/151271818-c97fdd79-1c36-49c4-a40a-a93b91afbfed.png)
